src/lib/consensus/proof_of_stake.ml

```diff
@@ -665,32 +667,47 @@ module Vrf = struct
 end
 
 module Epoch_data = struct
-  type ('epoch_ledger, 'epoch_seed, 'protocol_state_hash, 'length) t =
+  type ('epoch_ledger, 'epoch_seed, 'protocol_state_hash, 'length, 'amount) t =
     { ledger: 'epoch_ledger
     ; seed: 'epoch_seed
     ; start_checkpoint: 'protocol_state_hash
     ; lock_checkpoint: 'protocol_state_hash
-    ; length: 'length }
+    ; length: 'length
+    ; participated_stake: 'amount }
```

Comment:
We only need to track this for the current epoch, so we should probably only store this field once. Two epoch data records are stored in the consensus state.

Reply:
Do we not need the one from the last epoch data, that has the total stake? The one in the current epoch data would have the stake only up till the block in the current epoch.

Reply:
Confusion regarding this was resolved in a meeting. I was wrong, this is the correct way to implement this.